Replace color spectrum with react-colorful

nathanbrud opened the pull request:
## What this PR does
This PR replaces the color spectrum Iris implementation with [react-colorful package](https://omgovich.github.io/react-colorful/).

The current color select is very laggy and buggy.
One of the biggest bugs is that dragging stops when cursor is outside the spectrum so users can't select white / black which is very basic.

## Screenshots & Recordings <!-- It's really helpful to give your reviewer some extra context - A picture is worth a thousand words after all. -->
https://user-images.githubusercontent.com/9106375/221582453-6f0a9a38-90ca-4946-9f81-c1effe52659d.mp4


## How it does that
A lot of deleted code from the ColorSpectrum component and replaced with react-colorful.
Some unused props deleted as well from the ColorSelect state.
**No breaking changes at all - all props work the same**

## Testing
Run storybook and see all variants ColorSelect can be used in.

nathanbrud committed:
Replace color spectrum with react-colorful

nathanbrud committed:
Added ColorSelect Tests

juliewongbandue commented:
Hey @nathanbrud, thanks for taking this on. Before we make such a large change, we should definitely add a test here.

nathanbrud commented:
@juliewongbandue Sure, added test 💪

nathanbrud committed:
Fixed warning

nathanbrud committed:
Add package using yarn 1.x

Update yarn.lock

nathanbrud committed:
Code cleanup

juliewongbandue reviewed (changes requested):
Hey Nathan! Thanks (again) for taking this on. A few other packages in our Vimeo repo use some of the props removed in this PR, so to avoid introducing breaking changes in those places, a solution would be to make this a `ColorSelectV2` component.

Since we did not have tests for the original component, it's hard to know if we would be breaking anything. Moving forward, to refactor existing components, we should include tests for the original component first (with 100% coverage) in a separate PR. Then, changes can be made.

I just have some suggestions on the test for _this_ approach to the component though!

cc: @tylerthegrove

nathanbrud commented on src/components/inputs/ColorSelect/ColorSelect.test.tsx:
```diff
@@ -0,0 +1,105 @@
+describe('ColorSelect', () => {
+  it('Renders ColorSelect input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
```
Sure, changed to use aria-label

nathanbrud commented on src/components/inputs/ColorSelect/ColorSelectInput.tsx:
```diff
@@ -45,6 +45,7 @@ export function ColorSelectInput({
         style={style}
         type="text"
         value={HEX}
+        data-testid="color-select-input"
```
Sure, changed to use aria-label

nathanbrud commented on src/components/inputs/ColorSelect/ColorSelect.test.tsx:
```diff
@@ -0,0 +1,105 @@
+  it('Resets ColorSelect color using reset button', async () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
+    fireEvent.click(input);
```
Change all clicks to `userEvent`

nathanbrud commented on src/components/inputs/ColorSelect/ColorSelect.test.tsx:
```diff
@@ -0,0 +1,105 @@
+  it('Renders ColorSelect picker using input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
+    fireEvent.click(input);
```
Change all clicks to `userEvent`

nathanbrud committed:
Code review - use userEvent and remove data-testId

nathanbrud commented:
Hi @juliewongbandue, I changed the clicks to `userEvent` and removed all `data-testId`.

One of the benefits of using a third party here is that they write most of the tests for us - [as you can see here](https://github.com/omgovich/react-colorful/blob/master/tests/components.test.js)
And we can cover only the delta between the library and our component.

Regarding creating a new v2 component - I can do it if you like but I think we're missing the point here, the current component is very bad - users cannot select black/white in many places across Vimeo because dragging doesn't work when moving the mouse outside the spectrum.
So overriding the current component will fix a critical bug.

Your call :)
